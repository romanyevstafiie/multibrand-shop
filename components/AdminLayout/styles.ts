import styled from 'styled-components';

export const LayoutContainer = styled.section`
  display: grid;
  grid-template-columns: 1fr 8fr;
  min-height: 100vh;
`;

export const ChildrenWrapper = styled.main`
  background-color: ${({ theme }) => theme.dashboardGrayBg};
`;
