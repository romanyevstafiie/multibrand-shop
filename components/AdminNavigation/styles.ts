import styled from 'styled-components';

export const StyledAdminNavigation = styled.aside`
  display: flex;
  align-items: center;
  justify-content: center;
  position: sticky;
  top: 0;
  padding: 5rem 1rem;

  li {
    margin: 1rem 0;
  }
`;
