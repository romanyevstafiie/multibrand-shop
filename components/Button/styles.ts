import styled, { css, keyframes } from 'styled-components';
import { Props } from './Button';

const defaultButton = css`
  background: ${({ theme }) => theme.primaryGolden};
`;

const hoveredButton = css`
  background: ${({ theme }) => theme.secondaryGolden};
`;

const focusedButton = css`
  box-shadow: 0 0 0 3px ${({ theme }) => theme.secondaryGolden};
`;

const disabledButton = css`
  background: ${({ theme }) => theme.darkGray};
  cursor: not-allowed;
`;

const loadingKeyframes = keyframes`
  0% {
    transform: translateX(25px);
  }
  100% {
    transform: translateX(-20px);
  }
`;

const loadingButton = css`
  &::before {
    content: '';
    position: absolute;
    z-index: 1;
    top: 0;
    left: -100%;
    width: 300%;
    height: 100%;
    background: ${({ theme }) => theme.primaryGolden}
      repeating-linear-gradient(
        60deg,
        transparent,
        transparent 10px,
        #ffa500 10px,
        #ffc130 20px
      );
    animation: ${loadingKeyframes} 1s infinite linear;
  }
  & > span {
    opacity: 0.5;
  }
  cursor: wait;
`;

const completedButton = css`
  padding: 0 2rem;
  background: ${({ theme }) => theme.primaryGreen};
  cursor: default;
`;

export const StyledButton = styled.button<Props>`
  position: relative;
  overflow: hidden;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  height: 3rem;
  padding: 0 2rem;
  font-family: inherit;
  font-size: 1rem;
  text-transform: uppercase;
  font-weight: 700;
  background: transparent;
  border: 0;
  border-radius: ${({ theme }) => theme.borderRadius};
  color: ${({ theme }) =>
    theme.mode === 'dark' ? theme.primaryWhite : theme.secondaryBg};
  cursor: pointer;
  outline: none;
  & > span {
    position: relative;
    display: flex;
    align-items: center;
    z-index: 1;
  }
  ${(p) => !p.isLoading && defaultButton};
  &:hover {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && hoveredButton};
  }
  &:focus {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && focusedButton};
  }
  &:disabled {
    ${disabledButton};
  }
  ${(p) => p.isLoading && loadingButton};
  ${(p) => p.isCompleted && completedButton};
`;
