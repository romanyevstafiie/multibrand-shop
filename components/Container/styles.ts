import styled from 'styled-components';

export const StyledContainer = styled.section`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  max-width: ${({ theme }) => theme.maxWidth};
  margin: 0 auto;
  padding: 0 1rem;

  a {
    text-decoration: none;
  }
`;
