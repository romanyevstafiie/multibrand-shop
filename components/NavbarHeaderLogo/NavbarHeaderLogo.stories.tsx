import { Story } from '@storybook/react';
import React from 'react';
import NavbarHeaderLogo, { Props } from './NavbarHeaderLogo';

export default {
  title: 'navbar/NavHeaderLogo',
  component: NavbarHeaderLogo,
};

const Template: Story<Props> = (args) => <NavbarHeaderLogo {...args} />;

export const Logo = Template.bind({});
Logo.args = { headerText: 'Sneaker-maniacs' };
