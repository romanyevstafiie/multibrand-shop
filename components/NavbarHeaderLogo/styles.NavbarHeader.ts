import styled from 'styled-components';

export const HeaderLogo = styled.h1`
  margin-left: 1rem;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
`;

export const Sneaker = styled.a`
  font-size: 1.75rem;
  color: ${({ theme }) => theme.primaryWhite};
  text-transform: uppercase;
  margin-left: 0.25rem;
  ::after {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
  }
`;
export const Maniacs = styled.span`
  font-size: 1.75rem;
  color: ${({ theme }) => theme.primaryGolden};
  text-transform: uppercase;
`;
