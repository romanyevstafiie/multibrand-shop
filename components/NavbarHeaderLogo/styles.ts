import styled from 'styled-components';

export const StyledHeaderLogo = styled.h1`
  margin-left: 1rem;
`;
