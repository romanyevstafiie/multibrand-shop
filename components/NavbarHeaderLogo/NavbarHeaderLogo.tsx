import Image from 'next/image';
import Link from 'next/link';
import React from 'react';
import * as Styled from './styles.NavbarHeader';

/**
 * Renders navbar header logo
 *@function NavbarHeaderLogo
 *@returns {JSX.Element} - Rendered NavbarHeaderLogo
 */
const NavbarHeaderLogo = (): JSX.Element => {
  return (
    <Styled.HeaderLogo>
      <Image
        src="/sneaker-maniacs-logo.png"
        alt="sneaker maniacs logo"
        height={60}
        width={60}
        quality={100}
        objectFit="contain"
      />
      <Link href="/" passHref>
        <Styled.Sneaker>Sneaker</Styled.Sneaker>
      </Link>
      <Styled.Maniacs>Maniacs</Styled.Maniacs>
    </Styled.HeaderLogo>
  );
};

export default NavbarHeaderLogo;
