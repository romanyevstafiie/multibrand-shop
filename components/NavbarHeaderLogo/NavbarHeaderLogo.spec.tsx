import { shallow } from 'enzyme';
import 'jest-styled-components';
import Image from 'next/image';
import React from 'react';
import renderer from 'react-test-renderer';
import NavbarHeaderLogo from './NavbarHeaderLogo';
import * as Styled from './styles.NavbarHeader';

describe('Navbar Header Logo', () => {
  const navHeader = shallow(<NavbarHeaderLogo />);

  it('should render with no errors', () => {
    expect(navHeader);
    expect(navHeader.length).toBe(1);
  });

  it('should render logo img with correct attributes', () => {
    const logoImg = navHeader.find(Image);

    expect(logoImg.prop('alt')).toBe('sneaker maniacs logo');
    expect(logoImg.prop('src')).toBe('/sneaker-maniacs-logo.png');
    expect(logoImg.prop('height')).toBe(60);
    expect(logoImg.prop('width')).toBe(60);
  });

  it('should render correct logo text', () => {
    const sneaker = navHeader.find(Styled.Sneaker);
    expect(sneaker.text()).toBe('Sneaker');
    const maniacs = navHeader.find(Styled.Maniacs);
    expect(maniacs.text()).toBe('Maniacs');
  });
});

describe('Header logo matches snapshot', () => {
  const tree = renderer.create(<NavbarHeaderLogo />).toJSON();
  expect(tree).toMatchSnapshot();
});
