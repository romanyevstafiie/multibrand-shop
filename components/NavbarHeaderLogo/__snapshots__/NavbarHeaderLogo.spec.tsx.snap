// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
.c0 {
  margin-left: 1rem;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.c1 {
  font-size: 1.75rem;
  text-transform: uppercase;
  margin-left: 0.25rem;
}

.c1::after {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.c2 {
  font-size: 1.75rem;
  text-transform: uppercase;
}

<h1
  className="c0"
>
  <div
    style={
      Object {
        "boxSizing": "border-box",
        "display": "inline-block",
        "margin": 0,
        "maxWidth": "100%",
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "maxWidth": "100%",
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        role="presentation"
        src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjAiIGhlaWdodD0iNjAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
        style={
          Object {
            "border": "none",
            "display": "block",
            "margin": 0,
            "maxWidth": "100%",
            "padding": 0,
          }
        }
      />
    </div>
    <noscript>
      <img
        alt="sneaker maniacs logo"
        decoding="async"
        src="/_next/image?url=%2Fsneaker-maniacs-logo.png&w=128&q=100"
        srcSet="/_next/image?url=%2Fsneaker-maniacs-logo.png&w=64&q=100 1x, /_next/image?url=%2Fsneaker-maniacs-logo.png&w=128&q=100 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": "contain",
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
    <img
      alt="sneaker maniacs logo"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": "contain",
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </div>
  <a
    className="c1"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Sneaker
  </a>
  <span
    className="c2"
  >
    Maniacs
  </span>
</h1>
`;
